lib/searchkick/index.rb

```diff
@@ -146,34 +135,43 @@ def clean_indices
       indices
     end
 
+    def set_alias_to_index(alias_name, index_name, clear=true)
+      old_indices = begin
+        client.indices.get_alias(name: alias_name).keys
+      rescue Elasticsearch::Transport::Transport::Errors::NotFound
+        []
+      end
+
+      actions = clear ? old_indices.map { |old_name| {remove: {index: old_name, alias: alias_name}} } : []
+      actions += [{add: {index: index_name, alias: alias_name}}]
+
+      client.indices.update_aliases body: {actions: actions}
+    end
+
     # https://gist.github.com/jarosan/3124884
     # http://www.elasticsearch.org/blog/changing-mapping-with-zero-downtime/
-    def reindex_scope(scope, options = {})
+    def reindex_scope(scope, options ={})
       skip_import = options[:import] == false
 
-      clean_indices
+      write_alias = "#{name}_write"
+      read_alias = name
 
+      # This is the "new" index we are reindexing to
       index = create_index
 
-      # check if alias exists
-      if alias_exists?
-        # import before swap
-        index.import_scope(scope) unless skip_import
+      # set writes to go to new index and don't remove alias to old index
+      set_alias_to_index(write_alias, index.name, false)
 
-        # get existing indices to remove
-        swap(index.name)
-        clean_indices
-      else
-        delete if exists?
-        swap(index.name)
+      # set reads to go to new index if there is no existing read alias
+      set_alias_to_index(read_alias, index.name) unless alias_exists?(read_alias)
 
-        # import after swap
-        index.import_scope(scope) unless skip_import
-      end
+      index.import_scope(scope) unless skip_import
 
-      index.refresh
+      # set writes and reads to only go to new index
+      set_alias_to_index(write_alias, index.name)
+      set_alias_to_index(read_alias, index.name)
 
-      true
+      clean_indices
```

Comment:
should we `ensure` a cleanup if the import fails?  (Not sure.  We do in our code, I think.)